bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/link/LinkHandler.java

```diff
@@ -0,0 +1,56 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.commons.link;
+
+import org.apache.sling.api.resource.Resource;
+import org.jetbrains.annotations.NotNull;
+
+import com.day.cq.wcm.api.Page;
+
+/**
+ * Computes a link based on input configuration.
+ * This is a Sling model that can be injected into other models using the <code>@Self</code> annotation.
+ */
+public interface LinkHandler {
+
+    /**
+     * Returns a link builder based on the given resource.
+     *
+     * @param resource Resource to read the link properties from.
+     * @return {@link LinkBuilder}
+     */
+    @NotNull
+    LinkBuilder get(@NotNull Resource resource);
+
+    /**
+     * Returns a link builder pointing to the given target page.
+     *
+     * @param page Target page of the link.
+     * @return {@link LinkBuilder}
+     */
+    @NotNull
+    LinkBuilder get(@NotNull Page page);
+
+    /**
+     * Returns a link builder pointing to the URL.
+     *
+     * @param URL URL string of the link.
+     * @return {@link LinkBuilder}
+     */
+    @NotNull
+    LinkBuilder get(@NotNull String URL);
+
+}
```

Comment:
one use case is currently not covered by the interface: how to build a link **pointing to an asset** (e.g. PDF download)? there is no signature for it. it could be confused with `get(<Resource>)` by passing the resource pointing to the asset, but that will not work as the resource is only used to read the link properties from.
probably best to add a new method to avoid having to create a synthetic resource with a link property pointing to an asset.

alternatively `get(<String>)` could be made more magic by infer from the string if it is an external url, a page path or an asset path. parameter name and javadoc need to be updated in that case.

Reply:
good catch! I added a specific method to define a link based on an asset at https://github.com/adobe/aem-core-wcm-components/pull/2196/commits/91c774f82f21b3b8a5c3116ee3776f21783d4f59